lib/script/taproot.js

```diff
@@ -0,0 +1,99 @@
+/*!
+* taproot.js - taproot interpreter for bcoin
+* https://github.com/bcoin-org/bcoin
+*/
+
+'use strict';
+
+const bio = require('bufio');
+const assert = require('assert');
+const { strcmp } = require('../utils/util');
+
+const {
+  TapLeafHash,
+  TapBranchHash,
+  TapTweakHash
+} = require('../utils/taggedhash');
+
+/**
+ * TapLeaf
+ */
+
+class TapLeaf {
+  /**
+   * Create a leaf in taproot tree.
+   * @constructor
+   * @param {Script} script
+   * @param {Number} leafVersion
+   */
+
+  constructor(script, leafVersion) {
+    assert(script);
+    assert(leafVersion !== undefined);
```

Comment:
why not just this?

```suggestion
    assert(leafVersion);
```

Reply:
oh right sorry, this should be `!= null` then:

```
> 0 != null
true
> null != null
false
> undefined != null
false
```

Reply:
should we remove this assert altogether?

Reply:
no i think is good to have, its probably ok as is, its just more the bcoin pattern to use `!= null`